src/Service/Models/GraphQLFilterParsers.cs

```diff
@@ -185,7 +185,7 @@ public GQLFilterParser(ISqlMetadataProvider metadataProvider)
                             // This path will only be exercised for CosmosDb_NoSql
                             queryStructure.DatabaseObject.Name = sourceName + "." + backingColumnName;
                             queryStructure.SourceAlias = sourceName + "." + backingColumnName;
-                            string? nestedFieldType = _metadataProvider.GetSchemaGraphQLFieldTypeByFieldName(queryStructure.EntityName, name);
+                            string? nestedFieldType = _metadataProvider.GetSchemaGraphQLFieldTypeByEntityFieldName(queryStructure.EntityName, name);
```

Comment:
```suggestion
                            string? nestedFieldObjectType = _metadataProvider.GetSchemaGraphQLFieldTypeByEntityFieldName(queryStructure.EntityName, name);
```

This naming would (in my opinion) clarify what this variable represent. I needed to double check, but this phrasing "Object Type" aligns with GraphQL specification https://spec.graphql.org/October2021/#sel-FAHZPABDBzBu_X

Reply:
This method is also able to get the scalar type such as ID, Int, etc. Naming ObjectType is not quite reflecting that.